File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/PayrollColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { Month } from "./Month";
import { PayrollRow } from "./PayrollRow";

export interface PayrollColumns {
    PayrollId: Column<PayrollRow>;
    EmployeeFullName: Column<PayrollRow>;
    Month: Column<PayrollRow>;
    Year: Column<PayrollRow>;
    BasicSalary: Column<PayrollRow>;
    Allowances: Column<PayrollRow>;
    Deductions: Column<PayrollRow>;
    NetSalary: Column<PayrollRow>;
    GeneratedDate: Column<PayrollRow>;
}

export class PayrollColumns extends ColumnsBase<PayrollRow> {
    static readonly columnsKey = 'Operations.Payroll';
    static readonly Fields = fieldsProxy<PayrollColumns>();
}

[Month]; // referenced types

File: HRMS/HRMS.Web/Modules/Operations/Payroll/PayrollColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;
using HRMS.Operations;

namespace HRMS.Operations.Columns;

[ColumnsScript("Operations.Payroll")]
[BasedOnRow(typeof(PayrollRow), CheckNames = true)]
public class PayrollColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int PayrollId { get; set; }
    [EditLink]
    public string EmployeeFullName { get; set; }
    [Width(100), QuickFilter]
    public Month Month { get; set; }
    [Width(80), QuickFilter]
    public int Year { get; set; }
    public decimal BasicSalary { get; set; }
    public decimal Allowances { get; set; }
    public decimal Deductions { get; set; }
    public decimal NetSalary { get; set; }
    public DateTime GeneratedDate { get; set; }
}
